### ISSUE TITLE
Fetch TO-DOs using a query library instead of manual control

### ISSUE THREAD

**Issue by Danswar** [OM33/to-do-app#7]
As we continue to move toward to more modern ways to make React code, let's refactor one more time our logic for fetching to-dos so we can meet more standard patterns.

What we want this time is to remove our manual control on the states of the request to the server (e.g: [task state](https://github.com/OM33/to-do-app/blob/master/src/components/TodoForm.js#L15), [loading state](https://github.com/OM33/to-do-app/blob/master/src/components/TodoForm.js#L16), [error state](https://github.com/OM33/to-do-app/blob/master/src/components/TodoForm.js#L17), [fetching on mount](https://github.com/OM33/to-do-app/blob/master/src/components/TodoForm.js#L34), [toggle loading](https://github.com/OM33/to-do-app/blob/master/src/components/TodoForm.js#L31), etc) and delegate that into a query library. This will potentially simplify our code base, while at the same time, will make our code more robust as it relies on production-tested third party libraries.

Feel free to use any query library of your preference, some examples could be: [React query](https://tanstack.com/query/latest), [SWR](https://swr.vercel.app) or any other similar.

**Comment by Danswar** [comment]
Please feel free to ask any question, or if an introduction to those technologies is required. I'll be happy to assist you.

**Comment by OM33** [comment]
Hi! great, actually I'm not sure how to proceed with this task. Could you give me some guidance? I would appreciate this.

**Comment by Danswar** [comment]
Of course! Basically we want to use one of those libraries to fetch the TODOs from the API. 

As you can see in [this example offered by SWR](https://swr.vercel.app/#overview) with just one hook they allow us to fetch data, to know if it is loading or if perhaps there was an error, without the necessity of implementing this on our own using useEffect or thing like that.

This is a refactoring task so some parts of the code you wrote in the previous task still will be useful during this one and I think the most of the work will be in this https://github.com/OM33/to-do-app/blob/master/src/components/TodoForm.js
